src/benchmarks/micro/libraries/Microsoft.Extensions.DependencyInjection/ServiceProviderEngineBenchmark.cs

```diff
@@ -5,11 +5,14 @@
 using System;
 using System.Runtime.CompilerServices;
 using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
 
 namespace Microsoft.Extensions.DependencyInjection.Performance
 {
+    [BenchmarkCategory(Categories.Libraries)]
     public class ServiceProviderEngineBenchmark
     {
+#if INTERNAL_DI
```

Comment:
Disabling the code blobs that need to use internal `Microsoft.Extensions.DependencyInjection.ServiceProviderMode` for now with the INTERNAL_DI ifdefs.

This way I can have this PR bring over all perf tests and work on enabling these ones in a separate PR.

@JunTaoLuo are you fine with that?

Reply:
Once PR https://github.com/dotnet/runtime/pull/38012 gets merged and we can consume packages with those changes here, then the ifdefs here can be removed as they are not needed anymore. 

In the meantime I think this PR can be merged once approved.

Reply:
File an issue to track the work if it's not completed in this PR.

Reply:
Keeping track on this issue https://github.com/dotnet/runtime/issues/37821